libprecompiled/PrecompiledGas.cpp

```diff
@@ -24,24 +24,33 @@
 #include <libdevcore/Log.h>
 using namespace dev;
 using namespace dev::precompiled;
-
 std::map<InterfaceOpcode, int64_t> GasMetrics::OpCode2GasCost = {{InterfaceOpcode::EQ, VeryLow},
     {InterfaceOpcode::GE, VeryLow}, {InterfaceOpcode::GT, VeryLow}, {InterfaceOpcode::LE, VeryLow},
-    {InterfaceOpcode::NE, VeryLow}, {InterfaceOpcode::Limit, VeryLow},
-    {InterfaceOpcode::GetInt, VeryLow}, {InterfaceOpcode::GetAddr, VeryLow},
-    {InterfaceOpcode::Set, VeryLow}, {InterfaceOpcode::GetByte32, 32 * VeryLow},
-    {InterfaceOpcode::GetByte64, 64 * VeryLow}, {InterfaceOpcode::CreateTable, High},
-    {InterfaceOpcode::OpenTable, Low}, {InterfaceOpcode::Insert, High},
-    {InterfaceOpcode::Update, High}, {InterfaceOpcode::Delete, Mid}};
+    {InterfaceOpcode::LT, VeryLow}, {InterfaceOpcode::NE, VeryLow},
+    {InterfaceOpcode::Limit, VeryLow}, {InterfaceOpcode::GetInt, VeryLow},
+    {InterfaceOpcode::GetAddr, VeryLow}, {InterfaceOpcode::Set, VeryLow},
+    {InterfaceOpcode::GetByte32, 32 * VeryLow}, {InterfaceOpcode::GetByte64, 64 * VeryLow},
+    {InterfaceOpcode::GetString, 64 * VeryLow}, {InterfaceOpcode::CreateTable, High},
+    {InterfaceOpcode::OpenTable, Low}, {InterfaceOpcode::Select, Low},
+    {InterfaceOpcode::Insert, High}, {InterfaceOpcode::Update, High},
+    {InterfaceOpcode::Remove, Mid}};
 
 void PrecompiledGas::appendOperation(InterfaceOpcode const& _opType, unsigned const& _opSize)
 {
     m_operationList->push_back(std::make_pair(_opType, _opSize));
 }
 
-u256 PrecompiledGas::calTotalGas(uint64_t const& _maxMemSize)
+void PrecompiledGas::updateMemUsed(uint64_t const& _newMemSize)
+{
+    if (_newMemSize > m_memUsed)
+    {
+        m_memUsed = _newMemSize;
+    }
+}
+
+u256 PrecompiledGas::calTotalGas()
 {
-    return (calComputationGas() + calMemGas(_maxMemSize));
+    return (calComputationGas() + calMemGas(m_memUsed));

```

Comment:
unnecessary to pass member variable to a member function

Reply:
Fixed.